src/main/java/org/broadinstitute/hellbender/tools/walkers/variantutils/ReblockGVCF.java

```diff
@@ -0,0 +1,453 @@
+        headerLines.add(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.RAW_QUAL_APPROX_KEY));
+        headerLines.add(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.VARIANT_DEPTH_KEY));
+        headerLines.add(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.MAPPING_QUALITY_DEPTH));
+
+        // Remove GCVFBlocks
+        headerLines.removeIf(vcfHeaderLine -> vcfHeaderLine.getKey().startsWith(GVCF_BLOCK));
+
+        headerLines.remove(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.DOWNSAMPLED_KEY));
+        headerLines.remove(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.HAPLOTYPE_SCORE_KEY));
+        headerLines.remove(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.INBREEDING_COEFFICIENT_KEY));
+        headerLines.remove(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.MLE_ALLELE_COUNT_KEY));
+        headerLines.remove(GATKVCFHeaderLines.getInfoLine(GATKVCFConstants.MLE_ALLELE_FREQUENCY_KEY));
+
+
+        if ( INCLUDE_NON_VARIANTS ) {
+            // Save INFO header names that require alt alleles
+            for ( final VCFHeaderLine headerLine : headerLines ) {
+                if (headerLine instanceof VCFInfoHeaderLine ) {
+                    if (((VCFInfoHeaderLine) headerLine).getCountType() == VCFHeaderLineCount.A) {
+                        infoHeaderAltAllelesLineNames.add(((VCFInfoHeaderLine) headerLine).getID());
+                    }
+                }
+            }
+        }
+
+        if ( dbsnp.dbsnp != null  ) {
+            VCFStandardHeaderLines.addStandardInfoLines(headerLines, true, VCFConstants.DBSNP_KEY);
+        }
+
+        VariantContextWriter writer = createVCFWriter(outputFile);
+
+        try {
+            vcfWriter = new GVCFWriter(writer, GVCFGQBands, genotypeArgs.samplePloidy);
+        } catch ( IllegalArgumentException e ) {
+            throw new CommandLineException.BadArgumentValue("GQBands", "are malformed: " + e.getMessage());
+        }        vcfWriter.writeHeader(new VCFHeader(headerLines, inputHeader.getGenotypeSamples()));
+
+        logger.info("Notice that the -ploidy parameter is ignored in " + getClass().getSimpleName() + " tool as this is automatically determined by the input variant files");
+    }
+
+    // get VariantContexts from input gVCFs and regenotype
+    public void apply(VariantContext variant, ReadsContext reads, ReferenceContext ref, FeatureContext features) {
+        final VariantContext newVC = regenotypeVC(ref, variant);
+        if (newVC != null) {
+            vcfWriter.add(newVC);
+        }
+    }
+
+    /**
+     * Re-genotype (and re-annotate) a VariantContext
+     *
+     * @param ref            the ref context
+     * @param originalVC     the combined genomic VC
+     * @return a new VariantContext or null if the site turned monomorphic and we don't want such sites
+     */
+    protected VariantContext regenotypeVC(final ReferenceContext ref, final VariantContext originalVC) {
```

Comment:
Is this method named / documented correctly?  It seems more like reblocking them with some minimal stripping out of alleles rather than regenotyping.